core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -801,6 +802,24 @@ public void delete() throws IOException, InterruptedException {
                     }
                 }
             }
+            if (this instanceof ItemGroup) {
+                // delete individual items first
+                // (disregard whether they would be deletable in isolation)
+                // JENKINS-34939: do not hold the monitor on this folder while deleting them
+                // (thus we cannot do this inside performDelete)
+                try (ACLContext oldContext = ACL.as2(ACL.SYSTEM2)) {
+                    for (Item i : ((ItemGroup<?>) this).getItems(TopLevelItem.class::isInstance)) {
+                        try {
+                            i.delete();
+                        } catch (AbortException e) {
+                            throw (AbortException) new AbortException(
+                                    "Failed to delete " + i.getFullDisplayName() + " : " + e.getMessage()).initCause(e);
```

Comment:
Perhaps. It is not normal to call `AbortException.initCause`, since the point of this class is that it suppresses a stack trace (including causes & suppressed). Whether the `initCause` was ever actually helpful in https://github.com/jenkinsci/cloudbees-folder-plugin/pull/95, I am not sure; neither `AbstractFolder.delete` nor `AbstractItem.performDelete` throw this exception type, so it may be dead code.